.github/workflows/test.yml

```diff
@@ -86,7 +87,9 @@ jobs:
         uses: hashicorp/setup-terraform@v2
         with:
           terraform_wrapper: false
-          terraform_version: '~1.0'
+          terraform_version: '~1'
```

Comment:
What are the intentions and implications here?

I'm not very familiar with this constraint syntax `~1`. I know `go-version` recognises `~> 1.0` which means

> _Allows only the rightmost version component to increment._

https://developer.hashicorp.com/terraform/language/expressions/version-constraints#-3

Would that be equivalent to just `~` here, without the `>`? If so - then does that mean we allow upgrades to v2? That would likely be unwanted?

Reply:
See moved completion conversation for continued conversation